test/addons/callback-scope/binding.cc

```diff
@@ -0,0 +1,39 @@
+#include "node.h"
+#include "v8.h"
+
+#include <assert.h>
+#include <vector>
+
+namespace {
+
+void RunInCallbackScope(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+
+  assert(args.Length() == 4);
+  assert(args[0]->IsObject());
+  assert(args[1]->IsNumber());
+  assert(args[2]->IsNumber());
+  assert(args[3]->IsFunction());
+
+  node::async_context asyncContext = {
+    args[1].As<v8::Number>()->Value(),
+    args[2].As<v8::Number>()->Value()
```

Comment:
Why this instead of `args[n]->NumberValue()`?

Reply:
I've generally viewed the `NumberValue()` approach to be the more "correct" approach but @addaleax is slowly convincing me that I'm wrong, especially when we know the input types going in.